Arriba/Arriba/Model/Partition.cs

```diff
@@ -258,7 +261,12 @@ private ushort[] FindOrAssignLIDs(DataBlock.ReadOnlyDataBlock values, int idColu
 
             // If the insert array matches types with the column then we can use the native type to do a direct assignment from the input array
             // to the column array.  If the types do not match, we need to fallback to object to allow the Value class to handle the type conversion
-            ITypedAddOrUpdateWorker worker = NativeContainer.CreateTypedInstance<ITypedAddOrUpdateWorker>(typeof(AddOrUpdateWorker<>), idColumnDataType);
+            ITypedAddOrUpdateWorker worker;
+            if (_cachedWorkers.TryGetValue(idColumnDataType, out worker) == false)
```

Comment:
Class instances can't be shared across partitions (since there is a class member that is used).  The caching would have to be a pool, not just a cache.